src/UsersModule/Components/UsersList/UsersList.tsx

```diff
@@ -1,7 +1,260 @@
+import { Button, Modal } from "react-bootstrap";
+import Pagination from "../../../SharedMoudule/Components/Pagination/Pagination";
 export default function UsersList() {
+  const [pagesArray, setPagesArray] = useState("")
+  const [isLoading, setIsLoading] = useState(true);
+  const [usersList, setUsersList] = useState([]);
+  const [usersId, setUsersId] = useState("");
+  const [isActivatedUser, setIsActivatedUser] = useState("");
+  const [showToggle, setShowToggle] = useState(false);
+  const [showView, setShowView] = useState(false);
+  const [userDetails, setUserDetails] = useState({});
+  const [pageNum, setPageNum] = useState(1)
+  const handleCloseView = () => setShowView(false);
+  const handleShowView = () => setShowView(true);
+
+  const handleCloseToggle = () => setShowToggle(false);
+  const handleShowToggle = () => setShowToggle(true);
+
+  async function getToggleActivatedEmployee() {
+    try {
+      const { data } = await axios.put(
+        `${baseUrl}/Users/${usersId}`,
+        {},
+        {
+          headers: {
+            Authorization: localStorage.getItem("userToken"),
+          },
+        }
+      );
+      getUsersList();
+      handleCloseToggle();
+      if (data.isActivated) {
+        toast.success("Active user 😍😍");
+      } else {
+        toast.success("No Active user ☹️☹️");
+      }
+    } catch (error) {
+      toast.error(
+        error?.response?.data?.message ||
+          "There was a mistake toggling user activation."
+      );
+    }
+  }
+
+  async function getUsersList() {
+    try {
+      const data  = await axios.get(
+        `${baseUrl}/Users?pageSize=10&pageNumber=${pageNum}`,
+        {
+          headers: {
+            Authorization: localStorage.getItem("userToken"),
+          },
+        }
+      );
+      console.log(data.data);
+      
+      setUsersList(data.data.data);
+      setPagesArray(Array(data.data?.totalNumberOfPages).fill().map((_, i) => i + 1));
+
+    } catch (error) {
+      toast.error(error?.response?.data?.message || "There's a mistake.");
+    }
+    setIsLoading(false);
+  }
+
+  useEffect(() => {
+    getUsersList();
+  }, [pageNum]);
+
+  //? ===========handle Date==============>>
+  const formatDate = (timestamp: string) => {
+    const date = new Date(timestamp);
+    const formattedDate = `${date.getDate()}-${
+      date.getMonth() + 1
+    }-${date.getFullYear()}`;
+    return formattedDate;
+  };
   return (
-    <div>UsersList</div>
-  )
+    <>
+      <div className="container-fluid d-flex flex-column   ">
+        <PagesHeader title={"Users"} />
+        {!isLoading ? (
+          <>
+            <Filter setUsersList={setUsersList} />
+            {usersList.length > 0 ? (
+              <div className="table-responsive w-100">
+                <table className="w-100   table text-center">
+                  <thead className=" p-4">
+                    <tr>
+                      <th>UserName</th>
+                      <th>Statues</th>
```

Comment:
You need to switch these two columns: statuses and Email
![image](https://github.com/SalmaAboArab/Project-Management-System/assets/64327685/8da6b8f8-ff7b-4c2a-8937-688ef9cba3d5)